packages/runtime/container-runtime/src/test/dataStoreContext.spec.ts

```diff
@@ -333,94 +339,80 @@ describe("Data Store Context Tests", () => {
                     getInitialGCSummaryDetailsFn,
                 );
             });
-
-            it("can correctly initialize and generate attributes", async () => {
-                dataStoreAttributes = {
-                    pkg: JSON.stringify(["TestDataStore1"]),
-                    snapshotFormatVersion: "0.1",
+            const pkgName = "TestDataStore1";
+
+            function testGenerateAttributes(disableIsolatedChannels: boolean, expected: WriteFluidDataStoreAttributes) {
```

Comment:
Can you add comments here explaining what the intent is and what this is testing. Basically, for each value of `disableIsolatedChannels` and `expected` it runs the three tests inside.